src/core/cff_parser.js

```diff
@@ -1580,11 +1594,42 @@ var CFFCompiler = (function CFFCompilerClosure() {
       }
       return this.compileIndex(charStringsIndex);
     },
-    compileCharset: function CFFCompiler_compileCharset(charset) {
-      let length = 1 + (this.cff.charStrings.count - 1) * 2;
-      // The contents of the charset doesn't matter, it's just there to make
-      // freetype happy.
-      let out = new Uint8Array(length);
+    compileCharset: function CFFCompiler_compileCharset(charset, numGlyphs,
+                                                        strings, isCIDFont) {
+      // Freetype requires the number of charset strings be correct and MacOS
+      // requires a valid mapping for printing.
+      let out;
+      let numGlyphsLessNotDef = numGlyphs - 1;
+      if (isCIDFont) {
+        // In a CID font, the charset is a mapping of CIDs not SIDs so just
+        // create an identity mapping.
+        out = new Uint8Array([
+          2, // format
+          0, // first CID upper byte
+          0, // first CID lower byte
+          (numGlyphsLessNotDef >> 8) & 0xFF,
+          numGlyphsLessNotDef & 0xFF,
+        ]);
+      } else {
+        let length = 1 + numGlyphsLessNotDef * 2;
+        out = new Uint8Array(length);
+        out[0] = 0; // format 0
+        let charsetIndex = 0;
+        let numCharsets = charset.charset.length;
+        for (let i = 1; i < out.length; i += 2) {
+          let sid = 0;
+          if (charsetIndex < numCharsets) {
+            let name = charset.charset[charsetIndex++];
+            sid = strings.getSID(name);
+            if (sid === -1) {
+              sid = 0;
+              warn(`Couldn't find ${name} in CFF strings`);
```

Comment:
@brendandahl This seem to have caused *a lot* of warning "spam" in the console even for the default `tracemonkey.pdf` file.
From a cursory look it appears that `charset.charset` only contains numbers, and not strings, for most PDF files.